dist/api_err.py

```diff
@@ -83,19 +83,29 @@ def __init__(self, name, value, desc, long_desc=None, **flags):
         setting.'''),
 ]
 
-# Update the #defines in the wiredtiger.in file.
-tmp_file = '__tmp_api_err' + str(os.getpid())
-tfile = open(tmp_file, 'w')
-skip = 0
-for line in open('../src/include/wiredtiger.in', 'r'):
-    if not skip:
-        tfile.write(line)
-    if line.count('Error return section: END'):
+sub_errors = [
+    Error('WT_NONE', -32000,
+    'last API call was successful', '''
+    This is some placeholder code that is used to check if multiple errors work.
+    It should be changed later.'''),
+    Error('WT_COMPACTION_ALREADY_RUNNING', -32001,
+    "Cannot reconfigure background compaction while it's already running", '''
+    This is some placeholder code that is used to check if multiple errors work.
+    It should be changed later.'''),
+    Error('WT_SESSION_MAX', -32002,
+    "out of sessions, configured for XXX (including internal sessions)", '''
+    This is some placeholder code that is used to check if multiple errors work.
+    It should be changed later.'''),
```

Comment:
I like the examples; they helped me understand where we are going and I understand that this is used as a placeholder to show how things should be updated. However, it looks like dead code now. I would rather have the example of how to extend the error code in a JIRA ticket or any suitable doc.

Reply:
I changed the placeholder text to some actual descriptions. I don't know a lot about each of the errors specifically, so please let me know if you want to change the descriptions further. To clarify, these codes are the actual ones that will be used, it's just that I had placeholder text for the long descriptions

Reply:
> I agree with Etienne that we shouldn't include error codes used exclusively for tests. Keep in mind that these error codes are part of WiredTiger's external interface, and having placeholders there isn't ideal.
> 
> It would be great if this change could also introduce one or more genuinely useful secondary error codes. However, if that's not feasible for now, it's perfectly fine to commit the PR without new error codes - they can be added later.
> 
> For tests, we should find an approach that minimizes interference with the API. For example, error code generation could be handled within the test itself as part of its execution.

Regarding the last bit about the tests, are the way my unit and python tests set up correct? I've tried to just pass in the error code itself to directly get the error message, rather than trying to actually recreate the error.

Reply:
> Regarding the last bit about the tests, are the way my unit and python tests set up correct? I've tried to just pass in the error code itself to directly get the error message, rather than trying to actually recreate the error.

That depends on what the test case is designed to verify. If it’s testing the ability to read the error code from the struct, then it’s perfectly fine.

Reply:
I am still voting to remove the dead/untested code so far. The tech design should have a section about testing where you will add relevant sub-error codes.
cc @jeremythorp

Reply:
Sorry, I think I misunderstood before. When you say dead/untested code, do you mean the other sub-level error codes that I added (`WT_COMPACTION_ALREADY_RUNNING ` and `WT_SESSION_MAX `)? The reason I added this was having multiple sub-level error codes would work when the script runs. I can remove them, or add some extra checks in the unit and python tests.

Reply:
No worries - I should have made it clearer!
If you are able to test them, then we can leave them. Otherwise, we should remove them.

Reply:
Just updated the testing to include those sub-level error codes as well

Reply:
Thank you - if the test is testing the ID of each error code, that looks good to me. Otherwise, if we are trying to actually get those errors, we are missing those tests.

Reply:
Yup, right now we are just testing the IDs of the error codes, we don't have the ability to actually get the error, that'll come in milestone 2